Fix:  Replaced Black, isort, Flake8 with Ruff and Updated dependencies

alikhere opened the pull request:
### Description

This pull request simplifies the tooling in the project by replacing **Black**, **isort**, **Flake8**, and **Pycodestyle** with **Ruff**. The following changes have been made:

- **Removed** the following unnecessary dependencies from the `dev` section of `pyproject.toml` as Ruff replaces them:
  - **isort**
  - **flake8**
  - **pycodestyle**
  - **pyflakes**
  - **pep8-naming**

- **Added** **Ruff** as a dev dependency (`ruff = "^0.11.0"`) to the `pyproject.toml` file. This consolidates both **linting** and **formatting** tasks under one tool.

- **Updated** the `pyproject.toml` file to configure **Ruff**:
  - Added **format settings** to use spaces for indentation and enforce single quotes:
    ```toml
    [tool.ruff]
    format = { indent-style = "space", quote-style = "single" }
    ```
  - Updated the **linting rules** to include errors (`E`), warnings (`W`), flake-style issues (`F`), and ignored imports (`I`):
    ```toml
    lint = { select = ["E", "F", "W", "I"] }
    ```
  - Configured **Ruff** to recognize **pretix** as a known first-party package for sorting imports:
    ```toml
    isort = { known-first-party = ["pretix"] }
    ```

- **Updated GitHub Actions Workflow**:
  - Replaced **isort** and **flake8** jobs with a single **Ruff** job in `.github/workflows/style.yml`.
  - Added the following command to run Ruff for linting and formatting:
    ```bash
    ruff check --fix . && ruff format .
    ```

- **No Functional Changes**:
  - This PR only updates code style and formatting. No functional changes have been made.



### Fixes

- This PR fixes issue **#556**.

hongquan commented on pyproject.toml:
```diff
@@ -160,3 +156,8 @@ version = {attr = "pretix.__version__"}
+[tool.ruff]
+format = { indent-style = "space" }
+lint = { select = ["E", "F", "W", "I"] }
+isort = { known-first-party = ["pretix"] }
```
Missing blank line at the end of file.

alikhere replied:
cool.

hongquan commented:
Sorry, we forgot to mention that single quote (`'`) is preferred.

hongquan commented on pyproject.toml:
```diff
@@ -111,16 +111,12 @@ dependencies = [
-            'pyflakes==3.2.*',
-            'flake8==7.1.*',
-            'pep8-naming==0.14.*',
+            'ruff==0.4.4',
```
Why this version?

alikhere replied:
I noticed that the latest version of Ruff is `0.11.0`, and it’s currently installed in the environment. Would it be okay to update the `pyproject.toml` to use `ruff = "^0.11.0"` to allow the latest version? This will ensure we’re using the most up-to-date features and bug fixes.

hongquan replied:
Yes.

hongquan reviewed (changes requested):
You need to update GH actions as well.

hongquan commented:
Hi @alikhere, the title of PR, please capitalize the beginning of sentence.
The common rule is that, any message for human to read, please apply grammar and spelling rule learned in school.

alikhere commented:
ok

alikhere commented:
@hongquan Could you please review the PR.

hongquan commented:
@alikhere The purpose of "style" GH Action is to check if submitted code conform to standard, so it should not fix or write the code.

Your GH Action command is writing changes to the code:

```sh
ruff check --fix . && ruff format .
```

hongquan commented:
Did you test your changes in your localhost?

alikhere commented:
> @alikhere The purpose of "style" GH Action is to check if submitted code conform to standard, so it should not fix or write the code.
> 
> Your GH Action command is writing changes to the code:
> 
> ```shell
> ruff check --fix . && ruff format .
> ```

sorry my bad, it should be 
```sh
        uff check .

alikhere commented:
> Did you test your changes in your localhost?

I tried to setup project locally using Dockerfile but getting error while buiding docker image.


![Screenshot from 2025-03-20 05-56-12](https://github.com/user-attachments/assets/31727efb-fadc-4500-b8f4-1e7e2a0e77b6)

hongquan commented:
Please configure Ruff to ignore F405 error in _settings.py_ files.
It would be nice if you can read those Ruff report and decide which errors can be ignored, which errors should be fixed later.

alikhere commented:
> Please configure Ruff to ignore F405 error in _settings.py_ files. It would be nice if you can read those Ruff report and decide which errors can be ignored, which errors should be fixed later.

I have tested the changes, and now ruff successfully ignores F405 in the relevant settings.py files.

hongquan commented:
Every time you make some changes, you should show the report so that I can see and determine what should do next.
How about this idea:

> It would be nice if you can read those Ruff report and decide which errors can be ignored, which errors should be fixed later.

Can you do or not?

alikhere commented:
> Every time you make some changes, you should show the report so that I can see and determine what should do next. How about this idea:
> 
> > It would be nice if you can read those Ruff report and decide which errors can be ignored, which errors should be fixed later.
> 
> Can you do or not?

Good idea, i can do that.

alikhere commented:
> Please configure Ruff to ignore F405 error in _settings.py_ files. It would be nice if you can read those Ruff report and decide which errors can be ignored, which errors should be fixed later.


# Update: Resolved Conflicts, Ignored F405, and Applied Formatting


## 1 Resolved Merge Conflicts
- New code is added to upstream development branch which caused conflicts with my branch so i fetch the latest changes and resolved the conflicts.

## 2  Ignored F405 in settings.py
### Before ruff was reporting F405 warnings in settings.py.

![Screenshot from 2025-03-21 16-49-24](https://github.com/user-attachments/assets/c662e2f6-f372-497f-91f0-ce6b519111fe)


### After that  i added the following rule to pyproject.toml
  ```toml
  [tool.ruff.lint.per-file-ignores]
  "*/settings.py" = ["F405"]
```
then ran ruff check . and verified that Ruff no longer reports F405 in settings.py.

![Screenshot from 2025-03-21 16-50-21](https://github.com/user-attachments/assets/83f9fef6-34aa-4822-be1f-946ac6dde2db)



## 3 Applied Ruff Formatting on Merged Files
Since merge conflicts introduced new code, I ran ruff format .
This applied Ruff's formatting rules (line length 120, single quotes, etc.) to the updated files.

All changes have been tested, and Ruff is now correctly ignoring F405 in settings.py.

hongquan commented:
@alikhere I wonder if you are just repeating ChatGPT answer or just put some thought to the work. The last comment https://github.com/fossasia/eventyay-tickets/pull/580#issuecomment-2746097052 is not related to what I suggested you to do. It is just a repetition of https://github.com/fossasia/eventyay-tickets/pull/580#issuecomment-2741393077.

alikhere commented:
> @alikhere I wonder if you are just repeating ChatGPT answer or you did put some personal thought to your work. The last comment [#580 (comment)](https://github.com/fossasia/eventyay-tickets/pull/580#issuecomment-2746097052) is not related to what I suggested you to do. It is just a repetition of [#580 (comment)](https://github.com/fossasia/eventyay-tickets/pull/580#issuecomment-2741393077).

Apologies for the misunderstanding. I initially thought you wanted a summary of the changes I made, rather than an analysis of the ruff report itself. I will now go through the report carefully identify which issues should be ignored or fixed.

mariobehling commented:
Hello @alikhere any updates? Are you still on this?

alikhere commented:
> Hello @alikhere any updates? Are you still on this?

Yeah,
I have resolved the merge conflicts and ran ruff format . as well as ruff check ..

After running ruff check ., I found the following issues:

E501: Line too long (129 > 120)

E722: Bare except block

E741: Ambiguous variable names (e.g., l, O, I)

I’ve shared the report with you. Could you please advise on how to proceed with fixing these issues?
Should I address these issues or ignore any of them based on the project’s needs?

hongquan commented:
> E501: Line too long (129 > 120)

These errors are easy to fix. Please help fix.

> E722: Bare except block
> E741: Ambiguous variable names (e.g., l, O, I)

These errors will not be quick to fix, because they require you to understand the surrounding code. Then let add them to ignore list.

alikhere commented:
@hongquan  Thanks for the feedback!

- I have fixed the E501 issues (long lines >120 chars).

- Added E722 and E741 to the ignore list in pyproject.toml as suggested.

hongquan commented:
@alikhere Please merge "development" branch to your branch to follow up with the latest changes in "development".

alikhere commented:
@hongquan  Merged

hongquan commented:
@alikhere Why did you add your .env file to the repository?

alikhere commented:
I added the .env file while setting up the project locally for testing. I'll remove it and update the PR.

mariobehling commented:
Docker build is failing. Please check.

hongquan commented:
@alikhere Could you please resolve conflict?

alikhere commented:
> @alikhere Could you please resolve conflict?

I was little busy, i will resolve and inform you.

alikhere commented:
@mariobehling @hongquan  Docker build fixed.

hongquan commented:
Please help resolve conflict.

mariobehling commented:
Good job. Thank @alikhere!
